st2-ask/notification_rule.yaml

```diff
@@ -0,0 +1,21 @@
+---
+name: "notify_slack_execution_change"
+pack: "st2"
+description: "Watches for an execution status of "awaits" for a certain action ref
+enabled: true
+
+trigger:
+  type: core.st2.generic.actiontrigger
+
+criteria:
+  trigger.action_ref:
+      type: equals
+      pattern : st2.ask
+  trigger.status:
+      type: equals
+      pattern : awaits
+
+action:
+  ref: slack.post_message
+  parameters:
+    message: '{{trigger.execution_id}} is awaiting an approval action'
```

Comment:
Regarding the first question, that's up to the user. IMO we don't need to prescribe how approvers get notified - we can provide some examples, but the idea of using rules is that it puts the st2 admin in control of routing these approval requests.

Regarding the second, please see "Satisfying an st2.ask Execution" in `st2-ask/overview.md`